cmd/edit/machinepool/cmd.go

```diff
@@ -142,32 +168,29 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 
-	var replicas int
-
 	// Editing the default machine pool is a different process
 	if machinePoolID == "default" {
-		replicas, err = getReplicas(cmd)
-		if err != nil {
-			reporter.Errorf("Expected a valid number of replicas: %s", err)
+		autoscaling, replicas, minReplicas, maxReplicas := getReplicas(cmd, reporter, machinePoolID,
+			cluster.Nodes().AutoscaleCompute())
+
+		if !autoscaling && replicas < 2 ||
+			(autoscaling && cmd.Flags().Changed("min-replicas") && minReplicas < 2) {
```

Comment:
We should check here for multi-az and compare against 3 instead of 2

Reply:
Fixed